contrib/kind.sh

```diff
@@ -1163,5 +1167,11 @@ if [ "$KIND_INSTALL_PLUGINS" == true ]; then
 fi
 if [ "$KIND_INSTALL_KUBEVIRT" == true ]; then
   install_kubevirt
-  install_kubevirt_ipam_controller
+  deploy_primary_udn_nad
+  deploy_passt_binary
+
+  install_cert_manager
```

Comment:
Right now IPAM repo does not deploy cert-manager at all - leaving that to the user consuming it. 
That makes some sense, as not all users need to manually deploy it (i.e. they get it "for free" from their cluster)

I suggest we keep that behavior as we want to give @maiqueb something fast and easy for him to develop on. Adding a script to ipam-ext so that users can deploy it from the ipam-ext repo can be a follow up, indeed, but let's not condition it for this PR.

Reply:
> Right now IPAM repo does not deploy cert-manager at all - leaving that to the user consuming it. That makes some sense, as not all users need to manually deploy it (i.e. they get it "for free" from their cluster)

Thats not correct, it doesnt deploy it because it vendor OVN k8s and use kind.sh
with DEPLOY_KUBEVIRT, which deploys kubevirt+ipam+cert

Bundling the cert-manger with the IPAM, will make OVN-k8s cleaner,
and it will need to just deploy cert-manager once it opt-out IPAM
in the future it will even allow to opt out Kubevirt and install custom one.

For users that don't opt-out same default behavior is kept, so there is no reason to not do the change in this PR, right ? (i.e just move cert-manager next to ipam)
This way we can do it in steps, where each step is self contained and clean.

Reply:
Way I see it, we have 2 options here:
1. will consist of the following:
- move cert-manager to ipam-ext repo (will have a separate Makefile target). 
- remove cert-manager install from all the places it is currently deployed manually (CNAO when bumped, OVNK, etc..).

OR as suggested currently on this PR
2. ipam-ext doesn't deploy cert-manager. ovn-k will always deploy cert-manager. 

@oshoval am I right?

Reply:
I would go with option 2 for now, and if needed, change it later on.

ipam-extensions shouldn't have an opinion of what certificate tool to use.

Reply:
@maiqueb this PR currently does exactly that.

Reply:
Option [1] is not correct, only repo that depends on cert-manager that is deployed by OVN K8s kind.sh
is ipam-extensions, so we should not touch any other beside that.
Option [3] is what i suggested [above](https://github.com/ovn-org/ovn-kubernetes/pull/4655#discussion_r1736292990), it is better than [2] but will require a little follow-up
Miguel is the Captain here.

For completeness note please i also suggested [here](https://github.com/ovn-org/ovn-kubernetes/pull/4655#discussion_r1735841627) a solution for the OPT-IN in case desired in the future.